spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/support/ipresolver/XForwardedRemoteAddressResolver.java

```diff
@@ -93,7 +93,7 @@ public InetSocketAddress resolve(ServerWebExchange exchange) {
 		Collections.reverse(xForwardedValues);
 		if (xForwardedValues.size() != 0) {
 			int index = Math.min(xForwardedValues.size(), maxTrustedIndex) - 1;
-			return InetSocketAddress.createUnresolved(xForwardedValues.get(index), 0);
+			return new InetSocketAddress(xForwardedValues.get(index), 0);
```

Comment:
Why this change?

Reply:
Otherwise the `InetSocketAddress::getAddress` returns null and the `XForwardedRemoteAddressResolver` has no effect.
Moreover, the `IpSubnetFilterRule::matches` only works for resolved addresses.